sql/generate_reports.sh

```diff
@@ -269,34 +286,22 @@ else
           fi
 
         elif [[ -n "$YYYY_MM_DD" ]]; then
-          # Even if doing a force run we only wanna run up until date given in case next month is mid-run as don't wanna get just desktop data
-          if [[ $(grep "httparchive.blink_features.usage" $query) && $LENS == "" ]]; then # blink needs a special join, different for lenses
-            date_join="yyyymmdd <= REPLACE(\"$YYYY_MM_DD\",\"_\",\"\")"
-          elif [[ $(grep "httparchive.blink_features.usage" $query) && $LENS != "" ]]; then # blink needs a special join, different for lenses
-            date_join="yyyymmdd <= CAST(REPLACE(\"$YYYY_MM_DD\",\"_\",\"-\") AS DATE)"
-            # Skip 2022_05_12 tables
-            date_join="${date_join} AND yyyymmdd != \"2022-05-12\""
-          elif [[ $metric != crux* ]]; then # CrUX is quick and join is more compilicated so just do a full run of that
-            # If a date is given, then only run up until then (in case next month is mid run as don't wanna get just desktop data)
-            date_join="SUBSTR(_TABLE_SUFFIX, 0, 10) <= \"$YYYY_MM_DD\""
+          # Even if doing a force run we only wanna run up until date given in case next month is mid-run as do not wanna get just desktop data
+          if [[ $metric != crux* ]]; then # CrUX is quick and join is more compilicated so just do a full run of that
+            # If a date is given, then only run up until then (in case next month is mid run as do not wanna get just desktop data)
+            date_join="date <= \"$DATE\""
             # Skip 2022_05_12 tables
-            date_join="${date_join} AND SUBSTR(_TABLE_SUFFIX, 0, 10) != \"2022_05_12\""
+            date_join="${date_join} AND date != \"2022-05-12\""
```

Comment:
And skipping logic also can be omitted.

Reply:
Looks like this (bad) data doesn't exist in the new datamodel so will remove this check before merging.